onnxruntime/python/onnxruntime_validation.py

```diff
@@ -56,3 +56,95 @@ def check_distro_info():
     else:
         warnings.warn('Unsupported platform (%s). ONNX Runtime supports Linux, macOS and Windows platforms, only.' %
                       __my_system__)
+
+
+def validate_build_package_info():
+    import_ortmodule_exception = None
+    try:
+        from onnxruntime.training.ortmodule import ORTModule # noqa
+        has_ortmodule = True
+    except ImportError:
+        has_ortmodule = False
+    except Exception as e:
+        # this may happen if Cuda is not installed, we want to raise it after
+        # for any exception other than not having ortmodule, we want to continue
+        # device version validation and raise the exception after.
+        import_ortmodule_exception = e
+        has_ortmodule = True
+
+    package_name = ''
+    version = ''
+    cuda_version = ''
+
+    if has_ortmodule:
+        try:
+            # collect onnxruntime package name, version, and cuda version
+            from .build_and_package_info import package_name
+            from .build_and_package_info import __version__ as version
+
+            try:
+                from .build_and_package_info import cuda_version
+            except: # noqa
+                pass
+
+            if cuda_version:
+                # collect cuda library build info. the library info may not be available
+                # when the build environment has none or multiple libraries installed
+                try:
+                    from .build_and_package_info import cudart_version
+                except: # noqa
+                    warnings.warn('WARNING: failed to get cudart_version from onnxruntime build info.')
```

Comment:
Should we put something in `build_and_package_info` when we're doing a CPU build so that we can short-circuit all the CUDA-related warnings?

Reply:
good point.
build_and_package_info.py is only generated in setup.py when it is building for ortmodule (which is not cpu build). 
In the code here, it only references build_and_package_info.py when has_ortmodule is True.
I put except here just in case there package build (setup.py) does something wrong. It shall neve happen unless there is a bug in setup.py.  (see def save_build_and_package_info(package_name, version_number, cuda_version)

Regarding code path. yes it is covered with CI tests (any import onnxruntime will trigger it). I am manually test it in training build. I am also manually testing (no training) gpu build in my another environment.